### PR TITLE
Fix XML parsing and move test resources to own folder

### PR DESCRIPTION
**Pull request by s9latimm**
related se-sic/VaRA/issues/770

### CODE REVIEW

File: lib/Feature/FeatureModelParser.cpp [hunk 1]
```diff
@@ -15,6 +15,10 @@ using std::make_unique;
 
 namespace vara::feature {
 
+std::string trim(const std::string &S) {
+  return llvm::StringRef(S).trim().str();
```
Comment (vulder) [inline lib/Feature/FeatureModelParser.cpp]:
```suggestion
std::string trim(llvm::StringRef S) {
  return S.trim().str();
```
Would it not be simpler to directly take a string ref if we, in call cases convert it to one anyhow?

File: unittests/resources/xml/test_trim.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vulder) [inline unittests/resources/xml/test_trim.xml]: Maybe we should name the file `test_fm_with_whitespaces.xml` with just trim it's not really clear why this file is special.

File: include/vara/Feature/FeatureModelTransaction.h [hunk 3]
```diff
@@ -682,7 +682,9 @@ class AddChild : public FeatureModelModification {
                          Parent);
     assert(C && P);
 
-    removeEdge(*C->getParent(), *C);
+    if (C->getParent()) {
+      removeEdge(*C->getParent(), *C);
+    }
```
Comment (vulder) [inline include/vara/Feature/FeatureModelTransaction.h:687]: Is this a special case where we remove something from root? (so: `C == Root`)?

**Review by vulder — APPROVED** [review]

File: include/vara/Feature/FeatureModelTransaction.h (hunk 3 above)
Comment (s9latimm) [reply include/vara/Feature/FeatureModelTransaction.h:687]: Yes, exactly that is the intention. Additionally we are now more robust regarding removal of parents beforehand, as they are reassigned in this step anyway.

**Review by vulder — APPROVED** [review]

**Pull request merged**